hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueHandler.java

```diff
@@ -823,19 +797,13 @@ ContainerCommandResponseProto handleGetSmallFile(
       return malformedRequest(request);
     }
 
-    // The container can become unhealthy after the lock is released.
-    // The operation will likely fail/timeout in that happens.
-    try {
-      checkContainerIsHealthy(kvContainer);
-    } catch (StorageContainerException sce) {
-      return ContainerUtils.logAndReturnError(LOG, sce, request);
-    }
-
     GetSmallFileRequestProto getSmallFileReq = request.getGetSmallFile();
 
     try {
       BlockID blockID = BlockID.getFromProtobuf(getSmallFileReq.getBlock()
           .getBlockID());
+      checkContainerIsHealthy(kvContainer, blockID, Type.GetSmallFile);
+      BlockUtils.verifyBCSId(kvContainer, blockID);
       BlockData responseData = blockManager.getBlock(kvContainer, blockID);
 
```

Comment:
BCSID verification will implicitly happen in BlockManager.getBlock() call for containerBCSID. Second time will be redundant check.

Reply:
OK.